src/components/PermissionItems.vue

```diff
@@ -0,0 +1,176 @@
+import { mapGetters, useStore } from 'vuex';
+import { PermissionService } from '@/services/PermissionService';
+import { showToast } from '@/utils';
+import { hasError } from '@/adapter';
+import { DateTime } from 'luxon';
+import emitter from '@/event-bus';
+
+export default defineComponent({
+  name: 'PermissionItems',
+  components: {
+    IonCard,
+    IonCardHeader,
+    IonCardSubtitle,
+    IonCardTitle,
+    IonCheckbox,
+    IonIcon,
+    IonItem,
+    IonItemDivider,
+    IonLabel,
+    IonSearchbar,
+    IonToggle,
+  },
+  computed: {
+    ...mapGetters({
+      query: 'permission/getQuery',
+      currentGroupPermissions: 'permission/getCurrentGroupPermissions',
+      currentGroup: "permission/getCurrentGroup",
+      currentPermissionsByGroupType: "permission/getCurrentPermissionsByGroupType"
+    })
+  },
+  methods: {
+    async updateQuery() {
+      await this.store.dispatch('permission/updateQuery', this.query)
+    },
+    async updatePermissionAssociation(event: any, permission: any) {
+      event.stopImmediatePropagation();
+
+      let resp = {} as any;
+      const payload = {
+        groupId: this.currentGroup.groupId,
+        permissionId: permission.permissionId
+      }
+
+      let currentPermissions = JSON.parse(JSON.stringify(this.currentGroupPermissions))
+
+      try {
+        if(permission.isChecked) {
+          const fromDate = this.currentGroupPermissions[permission.permissionId].fromDate
+
+          resp = await PermissionService.removeSecurityPermissionFromSecurityGroup({
+            ...payload,
+            thruDate: DateTime.now().toMillis(),
+            fromDate
+          })
+
+          if(hasError(resp)) {
+            throw resp.data;
+          }
+
+          delete currentPermissions[permission.permissionId]
+        } else {
+          const time = DateTime.now().toMillis()
+          const params = {
+            ...payload,
+            fromDate: time
+          }
+
+          resp = await PermissionService.addSecurityPermissionToSecurityGroup(params)
+
+          if(hasError(resp)) {
+            throw resp.data;
+          }
+
+          currentPermissions[permission.permissionId] = params
+        }
+
+        if(!hasError(resp)) {
+          showToast(translate("Security group permission association successfully updated."))
+          await this.store.dispatch('permission/updateCurrentGroupPermissions', { groupId: this.currentGroup.groupId, currentPermissions})
+          this.store.dispatch('permission/checkAssociated')
+          event.target.checked = !permission.isChecked
+        } else {
+          throw resp.data
+        }
+      } catch(err) {
+        showToast(translate("Failed to update security group permission association."))
+        console.error(err)
+      }
+    },
+    arePermissionsAvailable() {
+      let arePermissionsAvailable = false;
+
+      Object.values(this.currentPermissionsByGroupType).map((groupType: any) => {
```

Comment:
Use some function instead of using map, as if any one of the groupType has permissions we don't need to check other groupTypes.